### PR TITLE
SNO-227 setup jupyter server

### PR DESCRIPTION
**Pull request by jgreen-unai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,12 +18,60 @@ Official docs available [here](https://medcattrainer.readthedocs.io/en/latest/)
 
 If you have any questions why not reach out to the community [discourse forum here](https://discourse.cogstack.org/)
 
-# Snomed Annotation Trial
+# Snomed Annotation Trial Server Setup
```
Comment (ariannasg) [inline README.md:21]: Minor: There are a few grammar errors in the README, some words are repeated/missing/misspelt, etc. Can we pls pass the README through a grammar checker and correct the errors?

File: jupyter/jupyterhub_config.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    def start(self):
+        self.volumes[f'/home/ubuntu/data'] = {
+            'bind': f'/home/{self.user.name}/data',
+            'mode': 'rw',  # or ro for read-only
+        }
+        return super().start()
+
+
+# c.JupyterHub.spawner_class = 'dockerspawner.DockerSpawner'
```
Comment (ariannasg) [inline jupyter/jupyterhub_config.py]: Minor: can we remove these comments?

File: jupyter/jupyterhub_config.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# dummy for testing. Don't use this in production!
+# c.JupyterHub.authenticator_class = "dummy"
+# c.DummyAuthenticator.password = "test"
+
+# launch with docker
+c.JupyterHub.spawner_class = UnaiDockerSpawner
+c.JupyterHub.hub_ip = '0.0.0.0'
+
+
+c.JupyterHub.bind_url = 'http://172.26.0.15:8000'
```
Comment (ariannasg) [inline jupyter/jupyterhub_config.py]: Why are we using the instance private IP here and in the line below? 
Isn't safer to use the static IP that won't change even if the instance is restarted?

File: nginx/sites-enabled/medcattrainer [hunk 4]
```diff
@@ -7,7 +7,7 @@ server {
 server {
     listen 443 ssl http2;
     listen [::]:443 ssl http2;
-    server_name snomed-annotation-trial.com;
+    server_name <enter-server-name-here>;
```
Comment (ariannasg) [inline nginx/sites-enabled/medcattrainer:10]: Why aren't we using the actual server names in this file?

File: scripts/jupyterhub.service [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=MedCATtrainer service
```
Comment (ariannasg) [inline scripts/jupyterhub.service]: Minor: Can we pls change the description to clarify this is the JupyterHub service?

File: scripts/start_med_cat_trainer.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+cd /home/ubuntu/MedCATtrainer || exit
+
+systemctl stop nginx.service
```
Comment (ariannasg) [inline scripts/start_med_cat_trainer.sh:5]: Is the Nginx service expected to be always running? Just checking if this command would raise an error when running the script on a new instance from scratch. Maybe we need to add something to continue the script if that particular step fails.

File: nginx/sites-enabled/medcattrainer (hunk 4 above)
Comment (jgreen-unai) [reply nginx/sites-enabled/medcattrainer:10]: The Repo is public and I don't want to advertise our server location.

File: scripts/start_med_cat_trainer.sh (hunk 6 above)
Comment (jgreen-unai) [reply scripts/start_med_cat_trainer.sh:5]: In our case there is no error. There was an issue where the base install of Nginx was blocking the Nginx container from starting correctly. I have removed nginx for the VM but some reason or another it was still blocking. Stopping service means the dockerise nginx can run correctly.

File: jupyter/jupyterhub_config.py (hunk 3 above)
Comment (jgreen-unai) [reply jupyter/jupyterhub_config.py]: I have remove from the Repo as to not expose our network information.

File: scripts/start_med_cat_trainer.sh (hunk 6 above)
Comment (jgreen-unai) [reply scripts/start_med_cat_trainer.sh:5]: Just checked running the command is not a problem for our script

File: README.md (hunk 1 above)
Comment (jgreen-unai) [reply README.md:21]: I have gone through it again and fixed what I can see.

File: scripts/jupyterhub.service [hunk 7]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=JupyterHub Service
```
Comment (jgreen-unai) [inline scripts/jupyterhub.service:2]: Updated

File: nginx/sites-enabled/medcattrainer (hunk 4 above)
Comment (ariannasg) [reply nginx/sites-enabled/medcattrainer:10]: That's fair

**Review by ariannasg — APPROVED** [review]

**Pull request merged**